### PR TITLE
Solidity fuzzer: Add import stmt generator

### PR DESCRIPTION
**Pull request by bshastry**
Depends on #10855

### CODE REVIEW

File: test/tools/ossfuzz/SolidityGenerator.h [hunk 1]
```diff
@@ -66,24 +66,54 @@ struct GenerationProbability
...
-	template <typename T>
-	std::string operator()(T const& _t)
+	explicit TestState(std::shared_ptr<RandomEngine> _rand):
+		sourceUnitPaths({}),
+		currentSourceUnitPath({}),
+		rand(std::move(_rand))
+	{}
+	/// Adds @param _path to @name sourceUnitPaths updates
+	/// @name currentSourceUnitPath.
+	void addSourceUnit(std::string& _path)
```
Comment (leonardoalt) [inline test/tools/ossfuzz/SolidityGenerator.h]: Why a non-const ref?

File: test/tools/ossfuzz/SolidityGenerator.h [hunk 2]
```diff
@@ -66,24 +66,54 @@ struct GenerationProbability
...
 	{
-		return _t->generate();
+		sourceUnitPaths.insert(_path);
+		currentSourceUnitPath = _path;
 	}
+	/// Returns true if @name sourceUnitPaths is empty,
```
Comment (leonardoalt) [inline test/tools/ossfuzz/SolidityGenerator.h]:
```suggestion
	/// @returns true if @name sourceUnitPaths is empty,
```

File: test/tools/ossfuzz/SolidityGenerator.h [hunk 3]
```diff
@@ -189,6 +228,18 @@ class PragmaGenerator: public GeneratorBase
...
+class ImportGenerator: public GeneratorBase
+{
+public:
+	explicit ImportGenerator(std::shared_ptr<SolidityGenerator> _mutator):
+	       GeneratorBase(std::move(_mutator))
+	{}
+	std::string visit() override;
+	std::string name() override { return "Import generator"; }
+private:
+	static constexpr size_t s_selfImportInvProb = 101;
```
Comment (leonardoalt) [inline test/tools/ossfuzz/SolidityGenerator.h]: Why 101?

File: test/tools/ossfuzz/SolidityGenerator.h (hunk 1 above)
Comment (bshastry) [reply test/tools/ossfuzz/SolidityGenerator.h]: Fixed.

File: test/tools/ossfuzz/SolidityGenerator.h (hunk 3 above)
Comment (bshastry) [reply test/tools/ossfuzz/SolidityGenerator.h]: Added a comment and change weight.

**Review by leonardoalt — COMMENTED** [review]
Please squash

**Comment by bshastry** [comment]
Weird, I squashed and force pushed but github shows older commits

**Comment by bshastry** [comment]
> Weird, I squashed and force pushed but github shows older commits

Ah now it looks good :)

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**